src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/WithClangNoPlatforms.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.Build.Utilities;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	class NdkToolsWithClangNoPlatforms : NdkToolsWithClang
+	{
+		public NdkToolsWithClangNoPlatforms (string androidNdkPath, NdkVersion version, TaskLoggingHelper? log)
+			: base (androidNdkPath, version, log)
+		{
+			// NDK r22 removed the arch-prefixed `ld` executable. It provides instead `ld.gold` and `ld.bfd`
+			// We're going to use the former
+
+			NdkToolNames[NdkToolKind.Linker] = "ld.gold";
+		}
+
+		public override IEnumerable<int> GetSupportedPlatforms ()
+		{
+			// NDK r22 and newer no longer have a single platforms dir.  The API level directories are now found in per-arch
+			// subdirectories under the toolchain directory. We need to examine all of them and compose a list of unique
+			// API levels (since they are repeated in each per-arch subdirectory, but not all architectures have the
+			// same set of API levels)
+			var apiLevels = new HashSet<int> ();
+			string sysrootLibDir = GetToolchainLibDir ();
+			foreach (AndroidTargetArch targetArch in Enum.GetValues (typeof (AndroidTargetArch))) {
```

Comment:
This feels odd; why use `Enum.GetValues()` instead of `new[]{AndroidTargetArch.Arm, …}`?

Reply:
Old code, copy pasta